### PR TITLE
Kafka src desitation

### PR DESCRIPTION
**Pull request by matzew**
Fixes #

## Proposed Changes

  * KafkaSource sink now uses Destination
  *
  *

**Release Note**

```release-note
KafkaSource sink now uses Destination
```

### CODE REVIEW

File: kafka/source/pkg/reconciler/kafkasource.go [hunk 1]
```diff
@@ -75,7 +76,8 @@ type Reconciler struct {
 	receiveAdapterImage string
 	eventTypeLister     eventinglisters.EventTypeLister
 	kafkaLister         listers.KafkaSourceLister
-	sinkReconciler      *duck.SinkReconciler
+	sinkResolver     	*resolver.URIResolver
+	sinkReconciler   	*duck.SinkReconciler
```
Comment (vaikas-google) [inline kafka/source/pkg/reconciler/kafkasource.go]: I don't think this should be there anymore? sinkResolver should be the only thing you need?

File: kafka/source/config/300-kafkasource.yaml [hunk 2]
```diff
@@ -126,7 +126,30 @@ spec:
...
+                          minLength: 1
+                        kind:
+                          type: string
+                          minLength: 1
+                        name:
+                          type: string
+                          minLength: 1
+                    uri:
+                      type: string
+                      description: "the target URI. If ref is provided, this must be relative URI reference."
```
Comment (grantr) [inline kafka/source/config/300-kafkasource.yaml]: You probably want https://github.com/knative/eventing/blob/86b99cb7741934bca31e6b862356f62c43e57c1b/config/300-apiserversource.yaml#L89-L104 also

Comment (matzew) [reply kafka/source/config/300-kafkasource.yaml]: ok, @grantr will PR the same for couch as well - after these are in, ok ?

Comment (grantr) [reply kafka/source/config/300-kafkasource.yaml]: :+1:

Comment (lionelvillard) [reply kafka/source/config/300-kafkasource.yaml]: I'm on it

Comment (grantr) [reply kafka/source/config/300-kafkasource.yaml]: @lionelvillard I think we should also change `anyOf` to `oneOf`

**Pull request merged**